floodviz/map_utils.py

```diff
@@ -113,3 +50,48 @@ def projection_info(code, url):
     if req.status_code != 200:
         return None
     return req.text
+
+
+def site_dict(ids, url_prefix):
+    """
+    Retrieves data on a list of sites and returns it in a list of dicts
+    :param ids: list of site IDs
+    :param url_prefix: NWIS site url prefix as a string
+    :return: list of dictionaries describing the sites
+    note that huc_cd, along with some other fields, will not be included
+    """
+    if not ids:
+        print('No site IDs provided')
+        return []
+
+    # generate the string of site ids for the url
+    id_input_string = ",".join(ids)
+    url = url_prefix + "site/?format=mapper&sites=" + id_input_string + "&siteStatus=all"
+
+    req = requests.get(url)
+    if req.status_code != 200:
+        print('request to NWIS has failed')
+        return []
```

Comment:
I would prefer for a failure of the request that None is returned. That is what the function did originally. The caller of the function can then distinguish between no data (which could be a legitimate response) and not receiving a response at all.